deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java

```diff
@@ -259,6 +278,29 @@ boolean canDeploy(EnvironBean envBean, String host, AgentBean agentBean, String
         }
     }
 
+    boolean canDeployWithRule(String hostId, EnvironBean envBean) throws Exception{
+        String envId = envBean.getEnv_id();
+        String ruleId = envBean.getDeploy_rule_id();
+
+        try {
+            LOG.debug("verify active agents for host {} rule {}", hostId, ruleId);
+            DeployRuleBean deployRuleBean = deployRuleDAO.getById(ruleId);
+            String tagValue = deployRuleBean.getCondition_key();
+            String tagName = deployRuleBean.getCondition_value();
```

Comment:
Seems that these two lines should be switched:
            String tagName = deployRuleBean.getCondition_key();
            String tagValue = deployRuleBean.getCondition_value();

Is this a sign for a test :-)?

Reply:
thanks for looking at it, the purpose is to add a `constraint` deployment feature to our teletraan deploy. and this PR is to give others an idea of it, after the team approves with the design idea, will get to the test phrase.